From ec69b448f4b6296edb6b28ced5ecb710be82438b Mon Sep 17 00:00:00 2001
From: Michael Weghorn <m.weghorn@posteo.de>
Date: Tue, 13 Feb 2024 16:25:35 +0100
Subject: [PATCH] scp2: Take ENABLE_{KF,QT}6 into account

The qt6 and kf6 VCL plugins are also part
of the kde install set, so take
`ENABLE_QT6` and `ENABLE_KF6` into
account when deciding whether to install
it.

This makes packaging these VCL plugins work
when building with `--enable-qt6` and/or
`--enable-kf6`, but without `--enable-qt5`
and `--enable-kf5`.

Command that can be used for testing:

    make DESTDIR=/tmp/dummy distro-pack-install -o build -o check

(Then, check whether the "installation" in /tmp/dummy
contains the VCL plugins as expected.)

Thanks to Andreas Sturmlechner for reporting
the issue on IRC #libreoffice-dev on 2024-02-12.

Change-Id: Ieea67468e3388f8b7b66f53221574c6225516b44
Reviewed-on: https://gerrit.libreoffice.org/c/core/+/163313
Tested-by: Jenkins
Reviewed-by: Michael Weghorn <m.weghorn@posteo.de>
---
 scp2/InstallScript_setup_osl.mk | 2 +-
 scp2/Module_scp2.mk             | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/scp2/InstallScript_setup_osl.mk b/scp2/InstallScript_setup_osl.mk
index 6f7b58204225e..cf69165c1b60f 100644
--- a/scp2/InstallScript_setup_osl.mk
+++ b/scp2/InstallScript_setup_osl.mk
@@ -36,7 +36,7 @@ $(eval $(call gb_InstallScript_use_modules,setup_osl,\
 	$(if $(filter TRUE,$(ENABLE_EVOAB2) $(ENABLE_GIO) $(ENABLE_GTK3)),\
 		scp2/gnome \
 	) \
-	$(if $(filter TRUE,$(ENABLE_QT5) $(ENABLE_KF5) $(ENABLE_GTK3_KDE5)),\
+	$(if $(filter TRUE,$(ENABLE_QT5) $(ENABLE_QT6) $(ENABLE_KF5) $(ENABLE_KF6) $(ENABLE_GTK3_KDE5)),\
 		scp2/kde \
 	) \
 	$(if $(filter TRUE,$(ENABLE_ONLINE_UPDATE)),\
diff --git a/scp2/Module_scp2.mk b/scp2/Module_scp2.mk
index 25ae8167df561..651413fa2c17e 100644
--- a/scp2/Module_scp2.mk
+++ b/scp2/Module_scp2.mk
@@ -40,7 +40,7 @@ $(eval $(call gb_Module_add_targets,scp2,\
 	$(if $(filter TRUE,$(ENABLE_EVOAB2) $(ENABLE_GIO) $(ENABLE_GTK3)),\
 		InstallModule_gnome \
 	) \
-	$(if $(filter TRUE,$(ENABLE_QT5) $(ENABLE_KF5) $(ENABLE_GTK3_KDE5)),\
+	$(if $(filter TRUE,$(ENABLE_QT5) $(ENABLE_QT6) $(ENABLE_KF5) $(ENABLE_KF6) $(ENABLE_GTK3_KDE5)),\
 		InstallModule_kde \
 	) \
 ))
